### PR TITLE
update .github directory to be v1.11 branch specific

### PR DESCRIPTION
**Pull request by aanm**
Some workflows need to be changed to be v1.11 specific. Also other
workflows and configurations can be removed since they are only used on
the master branch.

To perform this task the `.github` directory from the previous stable
branch was copy pasted and there was a manual check of the diff between
the modified files and the files from `master` branch. Heuristically,
this meant removing all GH workflows that were triggered by
`issue_comment` and the ones that are exclusively cron jobs, as well as
modifying the remaining workflows to be specific for this v1.11 stable
branch.

Signed-off-by: André Martins <andre@cilium.io>

### CODE REVIEW

**Review by joestringer — APPROVED** [review]
:exploding_head: This looks like a lot of work. I'm going to mostly just assume you caught everything that needs updating here.

**Review by nbusseneau — APPROVED** [review]
Workflow changes LGTM. Was a good thing cause I noticed I had incorrectly set up the `cilium-v1.11-runtime-kernel-net-next` job to still be `cilium-v1.11-runtime-kernel-4.9` as on 1.10, but we changed that recently :)

File: .github/workflows/build-images-ci.yaml [hunk 1]
```diff
@@ -83,39 +74,11 @@ jobs:
...
-
-      # Import GitHub's cache build to docker cache
-      - name: Copy ${{ matrix.name }} Golang cache to docker cache
-        uses: docker/build-push-action@a66e35b9cbcf4ad0ea91ffcaf7bbad63ad9e0229
-        with:
-          context: /tmp/.cache/${{ matrix.name }}
-          file: ./images/cache/Dockerfile
-          push: false
-          platforms: linux/amd64
-          target: import-cache
```
Comment (pchaigno) [inline .github/workflows/build-images-ci.yaml:112]: Why was this removed?

File: .github/workflows/lint-bpf-checks.yaml [hunk 2]
```diff
@@ -30,6 +30,7 @@ jobs:
         uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
         id: changes
         with:
+          base: ${{ github.event.pull_request.base.sha || github.event.before }}
```
Comment (pchaigno) [inline .github/workflows/lint-bpf-checks.yaml]: Where is this coming from? Could it be something we removed for a legitimate reason in master and forgot to backport to v1.10?

File: .github/workflows/lint-build-commits.yaml [hunk 3]
```diff
@@ -84,6 +85,7 @@ jobs:
         uses: dorny/paths-filter@78ab00f87740f82aec8ed8826eb4c3c851044126
         id: test-tree
         with:
+          base: ${{ github.event.pull_request.base.sha }}
```
Comment (pchaigno) [inline .github/workflows/lint-build-commits.yaml]: Ditto.

**Review by pchaigno — CHANGES_REQUESTED** [review]
Couple questions below.

File: .github/workflows/lint-bpf-checks.yaml (hunk 2 above)
Comment (pchaigno) [reply .github/workflows/lint-bpf-checks.yaml]: This line was removed in https://github.com/cilium/cilium/commit/652da822debe3404ed0b287aa18e7d30d8c17783#diff-2540bd4e18ec73cb134ede1c7ce5126ce3f6f69cb391b91bf4506dc5d3548f1a, which we didn't backport to v1.10. We should remove this line and the other two in `.github/workflows/lint-build-commits.yaml`.

Comment (pchaigno) [reply .github/workflows/lint-bpf-checks.yaml]: Come to think of it, we should just apply https://github.com/cilium/cilium/pull/16311 again on this branch and squash the three commits I think.

File: .github/workflows/build-images-ci.yaml (hunk 1 above)
Comment (aanm) [reply .github/workflows/build-images-ci.yaml:112]: We don't use build caches on stable branches, that's why it was removed. There's a limit of 5GB for GH cache.

File: .github/workflows/lint-bpf-checks.yaml (hunk 2 above)
Comment (aanm) [reply .github/workflows/lint-bpf-checks.yaml]: I've used the v1.10 as reference. It's coming from [here](https://github.com/cilium/cilium/blob/v1.10/.github/workflows/lint-bpf-checks.yaml#L33)

Comment (pchaigno) [reply .github/workflows/lint-bpf-checks.yaml]: Yeah, I know. But you only did that because you don't want to reimplement all changes made when we branched to v1.10, right? So we want the workflows to be the same as master, with just the changes required in stable branches (e.g., removing cron triggers, changing trigger phrases, etc.). This isn't one of those changes; those lines were removed in master to shut a warning. Thus, I think we should remove them in v1.11 as well.

**Comment by aanm** [comment]
pushed the following diff:
```diff
diff --git a/.github/workflows/lint-bpf-checks.yaml b/.github/workflows/lint-bpf-checks.yaml
index 48465af8fc..88096fe51b 100644
--- a/.github/workflows/lint-bpf-checks.yaml
+++ b/.github/workflows/lint-bpf-checks.yaml
@@ -30,7 +30,6 @@ jobs:
         uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
         id: changes
         with:
-          base: ${{ github.event.pull_request.base.sha || github.event.before }}
           filters: |
             bpf-tree:
               - 'bpf/**'
diff --git a/.github/workflows/lint-build-commits.yaml b/.github/workflows/lint-build-commits.yaml
index 4c2cae5e02..08443038bd 100644
--- a/.github/workflows/lint-build-commits.yaml
+++ b/.github/workflows/lint-build-commits.yaml
@@ -64,7 +64,6 @@ jobs:
         uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
         id: bpf-tree
         with:
-          base: ${{ github.event.pull_request.base.sha }}
           filters: |
             src:
               - 'bpf/**'
@@ -85,7 +84,6 @@ jobs:
         uses: dorny/paths-filter@78ab00f87740f82aec8ed8826eb4c3c851044126
         id: test-tree
         with:
-          base: ${{ github.event.pull_request.base.sha }}
           filters: |
             src:
               - 'test/**'
```
